### PR TITLE
[BugFix] fix several correctness issues about cloud native pk index

### PR DESCRIPTION
**Pull request by luohaha**
## Why I'm doing:
I found two correctness issues about cloud native pk index when running PK deterministic random test:
1. When handle major compaction, we merge same keys only by versions, and it's not complete. For example, we have same keys cross two segments in same `Rowset`, they have same version but we still need the `max_rss_rowid` to decide the order of these keys.
2. When compaction merge several Rowsets with only del files, it won't generate output Rowset now, so we will loss the del files to rebuild the index. So we need to generate output Rowset when merge Rowsets with only del files.

## What I'm doing:
Fix these two issues.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/lake/meta_file.cpp [hunk 1]
```diff
@@ -255,7 +255,11 @@ void MetaFileBuilder::apply_opcompaction(const TxnLogPB_OpCompaction& op_compact
     }
 
     // add output rowset
-    if (op_compaction.has_output_rowset() && op_compaction.output_rowset().segments_size() > 0) {
+    if (op_compaction.has_output_rowset() &&
+        (op_compaction.output_rowset().segments_size() > 0 || !collect_del_files.empty())) {
+        // NOTICE: we need output rowset in two scenarios:
+        // 1. We have output segments after compactions.
+        // 2. We need del files to rebuild cloud native PK index.
```
Comment (wyb) [inline be/src/storage/lake/meta_file.cpp:262]: line 271
```
_tablet_meta->set_next_rowset_id(_tablet_meta->next_rowset_id() + std::max(1, rowset->segments_size()));
```

Comment (luohaha) [reply be/src/storage/lake/meta_file.cpp:262]: done

File: be/test/storage/lake/primary_key_publish_test.cpp [hunk 2]
```diff
@@ -1296,6 +1296,74 @@ TEST_P(LakePrimaryKeyPublishTest, test_index_rebuild_with_dels2) {
     EXPECT_EQ(kChunkSize, read_rows(tablet_id, version));
 }
 
+TEST_P(LakePrimaryKeyPublishTest, test_index_rebuild_with_dels3) {
+    if (!GetParam().enable_persistent_index ||
+        GetParam().persistent_index_type != PersistentIndexTypePB::CLOUD_NATIVE) {
+        return;
```
Comment (kevincai) [inline be/test/storage/lake/primary_key_publish_test.cpp:1302]: leverage `GTEST_SKIP()` to explicit that the case is skipped under the condition.

**Review by kevincai — APPROVED** [review]

File: be/test/storage/lake/primary_key_publish_test.cpp (hunk 2 above)
Comment (luohaha) [reply be/test/storage/lake/primary_key_publish_test.cpp:1302]: Will do next

**Review by wyb — APPROVED** [review]

**Pull request merged**